src/main/scala/org/ergoplatform/nodeView/wallet/DefaultBoxSelector.scala

```diff
@@ -75,20 +78,18 @@ object DefaultBoxSelector extends BoxSelector {
     if (pickBoxes(inputBoxes, externalFilter, balanceMet)) {
       //then we pick boxes until all the target asset amounts are met (we pick only boxes containing needed assets).
       //If this condition is satisfied on the previous step, we will do one extra check (which is not that much).
-      if (pickBoxes(inputBoxes, bc => externalFilter(bc) && bc.assets.exists { case (id, _) =>
+      if (pickBoxes(inputBoxes, box => externalFilter(box) && box.assets.exists { case (id, _) =>
         val targetAmt = targetAssets.getOrElse(id, 0L)
         lazy val currentAmt = currentAssets.getOrElse(id, 0L)
         targetAmt > 0 && targetAmt > currentAmt
       }, assetsMet)) {
-        subtractAssets(currentAssets, targetAssets)
+        val notNewTargetedAssets =
+          targetAssets.filterNot { case (id, _) => res.headOption.exists(bx => Algos.encode(bx.id) == id) }
```

Comment:
why head box is dedicated here?

Reply:
@catena2w, this is how we can define wether new assets are presented in the sequence  or not.